### PR TITLE
Update gRPC auth with interceptor scope example

### PR DESCRIPTION
**Pull request by JamesNK**
I haven't written docs before where the example starts off with "imagine a scenario...". Let me know if there is a better way to do this.

### CODE REVIEW

File: aspnetcore/grpc/authn-and-authz.md [hunk 1]
```diff
@@ -128,6 +128,45 @@ services
     });
 ```
 
+A gRPC interceptor can also be used to configure a bearer token. An advantage to using an interceptor is the client factory can be configured to create a new interceptor for each client. This allows an interceptor to be constructed from DI using scoped and transient services.
+
+Imagine an app has a user defined `ITokenProvider` which is scoped per-request, and clients injected into gRPC services and Web API controllers should call it to set the bearer token on all gRPC calls:
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/authn-and-authz.md:133]:
```suggestion
Consider an app that has a user defined `ITokenProvider` which is scoped per-request, and clients injected into gRPC services and Web API controllers should call it to set the bearer token on all gRPC calls:
```

**Review by Rick-Anderson — APPROVED** [review]
We use **Consider**  frequently.

File: aspnetcore/grpc/authn-and-authz.md [hunk 2]
```diff
@@ -128,6 +128,48 @@ services
     });
 ```
 
+A gRPC interceptor can also be used to configure a bearer token. An advantage to using an interceptor is the client factory can be configured to create a new interceptor for each client. This allows an interceptor to be constructed from DI using scoped and transient services.
+
+Imagine an app has:
```
Comment (JamesNK) [inline aspnetcore/grpc/authn-and-authz.md:133]:
```suggestion
Consider an app that has:
```

File: aspnetcore/grpc/authn-and-authz.md [hunk 3]
```diff
@@ -128,6 +128,48 @@ services
     });
 ```
 
+A gRPC interceptor can also be used to configure a bearer token. An advantage to using an interceptor is the client factory can be configured to create a new interceptor for each client. This allows an interceptor to be constructed from DI using scoped and transient services.
+
+Consider an app that has:
+* A user defined `ITokenProvider` for getting a bearer token which is scoped per-request.
```
Comment (JamesNK) [inline aspnetcore/grpc/authn-and-authz.md:134]:
```suggestion
* A user defined `ITokenProvider` for getting a bearer token. It is registered in DI with a scoped per-request lifetime.
```

File: aspnetcore/grpc/authn-and-authz.md [hunk 4]
```diff
@@ -128,6 +128,48 @@ services
     });
 ```
 
+A gRPC interceptor can also be used to configure a bearer token. An advantage to using an interceptor is the client factory can be configured to create a new interceptor for each client. This allows an interceptor to be constructed from DI using scoped and transient services.
```
Comment (JamesNK) [inline aspnetcore/grpc/authn-and-authz.md:131]:
```suggestion
A gRPC interceptor can also be used to configure a bearer token. An advantage to using an interceptor is the client factory can be configured to create a new interceptor for each client. This allows an interceptor to be [constructed from DI using scoped and transient services](/dotnet/core/extensions/dependency-injection#service-lifetimes).
```

File: aspnetcore/grpc/authn-and-authz.md [hunk 5]
```diff
@@ -128,6 +128,48 @@ services
...
+    {
+        o.Address = new Uri("https://localhost:5001");
+    })
+    .AddInterceptor<AuthInterceptor>(InterceptorScope.Client);
+```
+
+The preceeding code:
+* Defines `AuthInterceptor` which is constructed using the user defined `ITokenProvider`.
+* Registers the `GreeterClient` type with client factory.
+* Configures the `AuthInterceptor` for this client using `InterceptorScope.Client`. A new interceptor is created for each client instance, so when a client is created for a gRPC service or Web API controller the scoped `ITokenProvider` is injected into the interceptor.
```
Comment (JamesNK) [inline aspnetcore/grpc/authn-and-authz.md:171]:
```suggestion
* Configures the `AuthInterceptor` for this client using `InterceptorScope.Client`. A new interceptor is created for each client instance. When a client is created for a gRPC service or Web API controller, the scoped `ITokenProvider` is injected into the interceptor.
```

**Pull request merged**